crates/matrix-sdk/src/client/mod.rs

```diff
@@ -3319,24 +3323,130 @@ pub(crate) mod tests {
     }
 
     #[async_test]
-    async fn test_room_preview_for_banned_room_retrieves_local_room_info() {
+    async fn test_media_preview_config() {
         let server = MatrixMockServer::new().await;
         let client = server.client_builder().build().await;
 
-        let room_id = room_id!("!a-room:matrix.org");
+        server
+            .mock_global_account_data()
+            .ok(
+                client.user_id().unwrap(),
+                ruma::events::GlobalAccountDataEventType::MediaPreviewConfig,
+                json!({
+                    "media_previews": "private",
+                    "invite_avatars": "off"
+                }),
+            )
+            .mount()
+            .await;
 
-        // Make sure the summary endpoint is not called
-        server.mock_room_summary().ok(room_id).never().mount().await;
+        let (initial_value, stream) =
+            client.account().observe_media_preview_config().await.unwrap();
 
-        // We create a locally cached banned room
-        let banned_room = client.inner.base_client.get_or_create_room(room_id, RoomState::Banned);
+        assert_eq!(initial_value.invite_avatars, InviteAvatars::Off);
+        assert_eq!(initial_value.media_previews, MediaPreviews::Private);
+        pin_mut!(stream);
+        assert_pending!(stream);
 
-        // And we get a preview, no server endpoint was reached
-        let preview = client
-            .get_room_preview(room_id.into(), Vec::new())
-            .await
-            .expect("Room preview should be retrieved");
+        server
+            .mock_sync()
+            .ok_and_run(&client, |builder| {
+                builder.add_global_account_data_event(GlobalAccountDataTestEvent::Custom(json!({
+                "content": {
+                    "media_previews": "off",
+                    "invite_avatars": "on"
+                },
+                "type": "m.media_preview_config"
+                  })));
+            })
+            .await;
+
+        let next_value = stream.next().await.unwrap();
```

Comment:
There's `assert_next_eq!()` in the `stream_asserts` crate which combines the next and await call, making this less verbose.

Reply:
the problem is that the content type cannot be equated